src/netbase.cpp

```diff
@@ -1291,6 +1291,24 @@ CSubNet::CSubNet(const std::string &strSubnet, bool fAllowLookup)
         network.ip[x] &= netmask[x];
 }
 
+CSubNet::CSubNet(const CNetAddr &addr, bool fAllowLookup)
+{
+    valid = addr.IsValid();
+    if (!valid)
+        return;
+
+    //set all netmask bits to 1
+    memset(netmask, 255, sizeof(netmask));
+
+    std::vector<CNetAddr> vIP;
+    if (!LookupHost(addr.ToString().c_str(), vIP, 1, fAllowLookup))
```

Comment:
Thanks. But this seems a bit circuitous. Let's do just:
```
memset(netmask, 255, sizeof(netmask));
network = addr;
```
Also the fAllowLookup parameter is unnecessary. It should never be needed to look up anything when going from an already binary address.